pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentPushUtils.java

```diff
@@ -134,6 +135,8 @@ public static void pushSegments(SegmentGenerationJobSpec spec, PinotFS fileSyste
                       segmentName, controllerURI, e);
               throw e;
             }
+          } finally {
+            FileUtils.deleteQuietly(tarFile);
```

Comment:
this one assumes the tar file is on local disk, which may not be true always. E.g. the tar file could be on S3.
The path is `s3://<my-bucket>/<my-tar>.tar.gz`
In this case, you need to use PinotFs to delete the tar file.

Reply:
Thanks for the review and pointing out to use PinotFS. Fixed.